geomesa-blobstore/geomesa-blobstore-core/src/main/scala/org/locationtech/geomesa/blob/core/AccumuloBlobStore.scala

```diff
@@ -87,9 +114,9 @@ class AccumuloBlobStore(ds: AccumuloDataStore) {
     (value.get, filename)
   }
 
-  private def putInternal(file: File, id: String) {
-    val localName = file.getName
-    val bytes =  ByteStreams.toByteArray(Files.newInputStreamSupplier(file))
+  private def putInternal(file: File, id: String, params: Map[String, String]) {
+    val localName = getFileName(file, params)
+    val bytes = ByteStreams.toByteArray(Files.asByteSource(file).openBufferedStream())
 
     val m = new Mutation(id)
```

Comment:
sorry to go outside scope of PR but there needs to be a bw.flush() after the mutation is added in order to actually return  a 200 series response to the user. currently this could result in a mutation rejected exception or other err and never get persisted.